app/components/anchor/component.rb

```diff
@@ -7,16 +7,38 @@ def self.generate_id(base_name: name.demodulize.underscore.dasherize)
       "#{base_name}-#{SecureRandom.uuid}"
     end
 
-    def initialize(classes: nil, data: {}, id: nil, **kwargs)
-      @classes = classes
-      @data = data
-      @id = id
+    def initialize(**kwargs)
+      classes = add_variant_classes_if_available(
+        classes: kwargs.delete(:classes) || [],
+        variant: kwargs[:variant]
+      )
+      @wrapper_options = kwargs.merge(class: classes)
```

Comment:
`wrapper_options` didn’t read very clearly to me, I wonder if something more direct like `component_options` could work?

```suggestion
      @component_options = kwargs.merge(class: classes)
```